libs/math/include/math/TVecHelpers.h

```diff
@@ -497,6 +497,27 @@ class TVecFunctions {
         return v;
     }
 
+    friend inline constexpr VECTOR<T> MATH_PURE mix(VECTOR<T> x, VECTOR<T> y, T a) noexcept {
+        return x * (VECTOR<T>(1) - a) + y * a;
+    }
+
+    friend inline constexpr VECTOR<T> MATH_PURE lerp(VECTOR<T> x, VECTOR<T> y, T a) noexcept {
+        return mix(x, y, a);
+    }
+
+    // Note that the mix/lerp in <scalar.h> already covers the following two cases, but they are
+    // defined here in case users do not include <scalar.h>. We use VECTOR<T> for the third argument
+    // instead of a secondary template argument (e.g. typename U) in order to avoid a naming
+    // conflict with the mix/lerp that is defined in <scalar.h>.
+
+    friend inline constexpr VECTOR<T> MATH_PURE mix(VECTOR<T> x, VECTOR<T> y, VECTOR<T> a) noexcept {
+        return x * (VECTOR<T>(1) - a) + y * a;
```

Comment:
Note that `1` only works for `float`/`double`/`bool`. It won't work for any of the integer versions of `vec`. Either `lerp`/`mix` should be disabled for those or you should add a new `static constexpr` to all `TVec*` implementations that publishes the `lerp unit` (255 for integers? Should it be different for shorts/bytes/etc.?). What does GLSL do here?

Reply:
Recent versions of GLSL do support integer variants, but only when `t` is a vector-of-bool.  IMO they should've introduced a `select` function rather than adding yet another overload.

So, my vote would be to disallow integers here, if I can figure out how to do that in C++...

Reply:
Jesus I broke MSVC.  All I wanted to do was mix vectors in an upcoming PR 😭